### DISCUSSION TITLE
Prospect of a built-in aligner for JBrowse

### DISCUSSION THREAD

**Discussion by rbuels** [GMOD/jbrowse-components#3346] [Ideas]
It strikes me that a built-in general purpose alignment tool like BLAST might be a good thing to have in JBrowse.

Some initial use cases for this:

* user has a smallish (100-500 b) sequence from someplace else and they would like to quickly find that sequence in the assembly or reference sequence they are currently looking at
* user has a gene sequence from someplace else and they want to look for matches for it in the genome they are currently looking at

**Comment by rbuels** [comment]
Whatever we would want to build in would probably need to compile to WebAssembly. Which means we would probably need to do the webassembly port ourselves, which is fine.

**Comment by tberardini** [comment]
We have had requests for this from our user community (TAIR). We are retiring an older genome browser that has this functionality and people are reporting that they will really miss not being able to enter sequences (like a pair of PCR primers) into a search box and having them show up mapped to the matched chromosomal locations. 

I think this feature would be very well used if developed for JBrowse2.
